src/matter/interaction/InteractionClient.ts

```diff
@@ -81,13 +81,14 @@ export class InteractionClient {
 
     constructor(
         private readonly exchangeManager: ExchangeManager<MatterController>,
```

Comment:
I would recommend to add another layer to handle those reconnect.
messageExchangeProvider: () => Promise<MessageExchange>
and the recoonect logic is implemented in the provider

Reply:
Ok, @mfucciOk, check nwo ... Ithink this is the "best" place now ... we can chat about "naming" of that one new class.

In fact the "InteractionClientMessanger" is now handling this with the ExchangeProvider and so we do it on the "exchange.send" level, but just in case that we have the case that WE initialize the session. For InteractionServer (and also InteractionCLient for Subscriptionhandler" the exchanges are not build by us but initiated by an incoming connection, so here we can not do the retry.

WDYT?